File: backend/src/main/java/com/ssafy/db/entity/Pet/Pet.java
package com.ssafy.db.entity.Pet;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@Entity
//Pet의 기본 속성 - 직접 입력받는건 pet name밖에 없음
public class Pet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long pet_id;

    // 펫 주인의 닉네임
    @Column(name = "nickname")
    private String nickname;

    // 펫 이름
    private String pet_name;

    @Column(nullable = true)
    private int exp;

    @ColumnDefault("1")
    @Column(nullable = false)
    private int level;

    private LocalDate create_date;

    @ColumnDefault("false")
    private boolean is_graduate;

    @Column(nullable = true)
    private LocalDate graduate_date;
}

File: mobile/app/src/main/java/com/team/teamrestructuring/dto/Pet.kt
package com.team.teamrestructuring.dto

import com.google.gson.annotations.SerializedName
import java.time.LocalDate

data class Pet(
    @SerializedName("pet_id")
    var pet_id:Long,

    @SerializedName("nickname")
    var nickname:String,

    @SerializedName("pet_name")
    var pet_name:String,

    @SerializedName("exp")
    var exp :Int,

    @SerializedName("level")
    var level:Int,

    @SerializedName("create_date")
    var create_date:String,

    @SerializedName("graduate_date")
    var graduate_date:String?,

    @SerializedName("_graduate")
    var _graduate:Boolean
){
    override fun toString(): String {
        return "Pet(pet_id=$pet_id, nickname='$nickname', pet_name='$pet_name', exp=$exp, level=$level, create_date=$create_date, graduate_date=$graduate_date, _graduate=$_graduate)"
    }
}
